pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,177 @@
+func GenerateChangeResourceRecordSetsInput(p *v1alpha3.ResourceRecordSetParameters, action route53.ChangeAction) *route53.ChangeResourceRecordSetsInput {
+	var ttl *int64
+	if p.TTL == nil {
+		num := int64(300)
+		ttl = &num
+	} else {
+		ttl = p.TTL
+	}
+
+	resourceRecords := make([]route53.ResourceRecord, 0)
+	for _, r := range p.Records {
+		record := r
+		resourceRecords = append(resourceRecords, route53.ResourceRecord{
+			Value: &record,
+		})
+	}
+
+	resourceRecordSet := route53.ResourceRecordSet{
+		Name:            p.Name,
+		Type:            route53.RRType(aws.StringValue(p.Type)),
+		TTL:             ttl,
+		ResourceRecords: resourceRecords,
+	}
+	changeBatch := &route53.ChangeBatch{
+		Changes: []route53.Change{
+			{
+				Action:            action,
+				ResourceRecordSet: &resourceRecordSet,
+			},
+		},
+	}
+	return &route53.ChangeResourceRecordSetsInput{
+		HostedZoneId: p.ZoneID,
+		ChangeBatch:  changeBatch,
+	}
+}
+
+// GenerateObservation is used to produce v1alpha3.ResourceRecordSetObservation from
+// route53.ResourceRecordSet
+func GenerateObservation(r route53.ResourceRecordSet) v1alpha3.ResourceRecordSetObservation {
+
+	rType := string(r.Type)
+	resourceRecords := make([]*v1alpha3.ResourceRecord, 0)
+	for _, item := range r.ResourceRecords {
+		resourceRecords = append(resourceRecords, &v1alpha3.ResourceRecord{Value: item.Value})
+	}
+
+	o := v1alpha3.ResourceRecordSetObservation{
+		Name:            r.Name,
+		Type:            &rType,
+		TTL:             r.TTL,
+		ResourceRecords: resourceRecords,
+	}
+
+	return o
+}
+
+// IsUpToDate checks if object is up to date
+func IsUpToDate(p v1alpha3.ResourceRecordSetParameters, rrset route53.ResourceRecordSet) (bool, error) {
+	patch, err := CreatePatch(&rrset, &p)
+	if err != nil {
+		return false, err
+	}
+	return cmp.Equal(&v1alpha3.ResourceRecordSetParameters{}, patch), nil
+}
+
+// LateInitialize fills the empty fields in *v1alpha3.ResourceRecordSetParameters with
+// the values seen in route53.ResourceRecordSet.
+func LateInitialize(in *v1alpha3.ResourceRecordSetParameters, rrSet *route53.ResourceRecordSet) {
+	if rrSet == nil {
+		return
+	}
+
+	in.Name = awsclients.LateInitializeStringPtr(in.Name, rrSet.Name)
+
+	rrType := string(rrSet.Type)
+	in.Type = awsclients.LateInitializeStringPtr(in.Type, &rrType)
+
+	in.TTL = awsclients.LateInitializeInt64Ptr(in.TTL, rrSet.TTL)
+	if len(in.Records) == 0 && len(rrSet.ResourceRecords) != 0 {
+		in.Records = make([]string, len(rrSet.ResourceRecords))
+		for i, val := range rrSet.ResourceRecords {
+			in.Records[i] = aws.StringValue(val.Value)
+		}
+	}
+}
+
+// CreatePatch creates a *v1beta1.ResourceRecordSetParameters that has only the changed
+// values between the target *v1beta1.ResourceRecordSetParameters and the current
+// *route53.ResourceRecordSet
+func CreatePatch(in *route53.ResourceRecordSet,
+	target *v1alpha3.ResourceRecordSetParameters) (*v1alpha3.ResourceRecordSetParameters, error) {
```

Comment:
This is subjective, but I prefer to avoid wrapping these longer function declarations. It's hard to follow where the params end and the function body starts since they're at the same level of indentation. These lines are a little long, but I usually don't worry about that too much unless the line is getting to well over the 120 char mark.

Reply:
The respective and subsequent functions have been unwrapped as suggested.